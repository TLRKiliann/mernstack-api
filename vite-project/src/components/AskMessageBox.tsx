import React from 'react'
import { UserType } from '../models/usertype'
import './styleComponents/AskMessageBox.scss'

type Field = {
  value?: string
}

type Form = {
  invite: Field
}

interface AskMessageBoxProps {
  catchById: UserType
  form: Form
  options: Field[]
  handleInviteChoice: (e: React.ChangeEvent<HTMLInputElement>) => void
  handleInvitation: (e: React.MouseEvent<HTMLButtonElement>) => void
  handleClose: (e: React.MouseEvent<HTMLButtonElement>) => void
}

const AskMessageBox: React.FC = (props: {AskMessageBoxProps, Form, Field}) => {
  //console.log(props.catchById, 'props.users')
  return (
    <div
      data-testid="testAskBox"
      className="boolean--result"
    >
      <form
        className="form--invitation"
      >
        <label htmlFor="users" className="lbl--invite">
          Invite
          <img
            src={props.catchById?.img}
            width="50px"
            height="50px"
            className="img--invite"
            alt="no img panel ask invite"
          />
          <span className="span--invite">
            {props.catchById?.firstName}&nbsp;
          </span> to private chat :
        </label>

        <label className="lbl--choicechat">
          Would you like&nbsp;
          <select 
            data-testid="selectid"
            id="invite"
            name="invite" 
            value={props.form}
            onChange={(e) => props.handleInviteChoice(e)}
            className="select--invite"
          >
          {props.options.map((option) => (
            <option
              data-testid="select-option"
              key={option.value}
              name="invite"
              value={option.value}
              className="option--invite"
            >
              {option.label}
            </option>
          ))}
          </select>
          &nbsp;chat.
        </label> 
        
        <button
          data-testid="btnInvite"
          type="button"
          onClick={(e) => props.handleInvitation(e)}
          className="btn--invitation"
        >
          Invite
        </button>

        <button
          data-testid="btnClose"
          onClick={props.handleClose}
          className="btn--close"
        >
          X
        </button>
      </form>
    </div>
  )
}

export default AskMessageBox