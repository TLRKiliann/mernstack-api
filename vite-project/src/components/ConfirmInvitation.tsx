import React from 'react'
import './styleComponents/ConfirmInvitation.scss'

interface ConfirmInvitationProps {
  isChecked: boolean
  handleCheckBox: () => void
  handleValidInvitation: (e: React.MouseEvent<HTMLButtonElement>) => void
}

const ConfirmInvitation: React.FC = (props: ConfirmInvitationProps) => {
  console.log(props, "props")
  return (
    <div className="box--confirminvitation">
      <h4 className="confirm">Confirm invitation ?</h4>
      <label>
        <input
          type="checkbox"
          checked={props.isChecked}
          onChange={props.handleCheckBox}
          className="input--validationInvite"
        />
        yes
      </label>

      <button
        type="button"
        onClick={props.handleValidInvitation}
        className="btn--validationlast"
      >
        Validate
      </button>
    </div>
  )
}

export default ConfirmInvitation