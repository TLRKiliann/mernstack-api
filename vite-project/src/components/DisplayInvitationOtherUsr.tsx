import React from 'react'

interface DisplayInvitationOtherUsrProps {
  isCheckInvite: boolean
  handleSwitchBox: () => void
  handleInvitedResponse: (e: React.MouseEvent<HTMLButtonElement>) => void
}

const DisplayInvitationOtherUsr: React.FC = (props: DisplayInvitationOtherUsrProps) => {
  return (
    <div>
      <h4 className="invitation--display">You've recieved invitation from {props.username}.</h4>
      <h4 className="invitation--display">Are you ok for {} chat ?</h4>
      <label>
        <input
          type="checkbox"
          checked={props.isCheckInvite}
          onChange={props.handleSwitchBox}
        />
        yes
      </label>
      <button
        type="button"
        onClick={props.handleInvitedResponse}
      >
        Validate
      </button>
    </div>
  )
}

export default DisplayInvitationOtherUsr