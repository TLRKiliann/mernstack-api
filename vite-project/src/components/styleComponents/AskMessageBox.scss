/*Invitation message*/
$purple: rgba(195,87,116, 1);
$sandybrown: rgba(244,164,96, 1);
$whitesmoke: #f5f5f5;
$dark_moderate_red: #ae3c52;
$navy: #000080;
$font_junction: 'Junction', sans-serif;

.boolean--result {
  position: absolute;
  top: 0px;
  right: 0px;
  left: 0px;
  width: 550px;
  margin: auto;
  margin-top: 20%;
  display: flex;
  background: linear-gradient(180deg,
    $sandybrown, $purple);
  border-radius: 15px;
  z-index: 4;
  .form--invitation {
    padding-top: 20px;
    padding-bottom: 20px;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    .lbl--invite {
      margin-top: 20px;
      display: flex;
      flex-direction: row;
      align-items: center;
      font-family: $font_junction;
      font-size: 1.4rem;
      color: $purple;
      .img--invite {
        margin-left: 10px;
        border-radius: 50% 50%;
        object-fit: cover;
      }
      .span--invite {
        margin-left: 10px;
        color: #50bcfe;
        text-shadow: 2px 1px 1px $navy;
      }
    }
    .lbl--choicechat {
      margin-top: 20px;
      padding: 10px;
      display: flex;
      align-items: center;
      font-family: $font_junction;
      font-size: 1.4rem;
      .select--invite {
        padding-top: 3px;
        padding-bottom: 3px;
        font-size: 0.9rem;
        background: $sandybrown;
        border: 1px solid $purple;
        border-radius: 3px;
        color: $purple;
        .option--invite {
        }
      }
    }
    .btn--invitation {
      margin-top: 20px;
      padding: 5px 20px;
      font-size: 1.0rem;
      font-weight: bold;
      background: linear-gradient(180deg,
        $sandybrown, orangered, $dark_moderate_red);
      border: 1px solid $purple;
      border-radius: 7px;
      color: $whitesmoke;
      &:hover {
        background: $sandybrown;
        color: $dark_moderate_red;
      }
    }
    .btn--close {
      position: absolute;
      top: 10px;
      right: 10px;
      padding: 3px 7px;
      object-fit: cover;
      background: radial-gradient($sandybrown, $purple);
      border: 1px solid $sandybrown;
      border-radius: 100%;
      color: $purple;
      &:hover {
        background: $purple;
        color: $whitesmoke;
      }
    }
  }
}