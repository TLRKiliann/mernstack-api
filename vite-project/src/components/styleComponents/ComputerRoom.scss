.nextcomp--room {
  position: absolute;
  top: 0px;
  right: 0px;
  bottom: 0px;
  left: 0px;
  background: linear-gradient(160deg, purple,
    orange);
  z-index: 1;
}

.div--animationroomStyle {
  position: absolute;
  top: 0px;
  right: 0px;
  left: 0px;
  width: 85%;
  margin: auto;
  margin-top: 25%;
  display: flex;
  opacity: 0;

  animation-name: rotation--bg;
  animation-delay: 0s;
  animation-duration: 4s;
  animation-iteration-count: 1;
  z-index: 2;

  .rotation--roomstyle {
    margin: auto;
    font-family: "Junction", sans-serif;
    font-size: 2rem;
    background: linear-gradient(to right, #f32170,
    #ff6b08, #cf23cf, #eedd44);
    -webkit-text-fill-color: transparent;
    -webkit-background-clip: text;
    opacity: 0;

    animation-name: rotation--room;
    animation-delay: 0s;
    animation-duration: 3s;
    animation-iteration-count: 1;
    animation-timing-function: ease-in-out;
    animation-direction: linear;
  }
}

@keyframes rotation--bg {
  0% {
    opacity: 1;
  }
  50% {
    opacity: 1;
  }
  75% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}

@keyframes rotation--room {
  0% {
    transform: rotate3d(2, -2, 1, 0deg);
    opacity: 1;
  }
  100% {
    transform: rotate3d(-3, 2, -3, 360deg);
    opacity: 1;
  }
}

.title--room {
  width: 90%;
  margin: auto;
  margin-top: 100px;
  margin-bottom: 20px;
  color: orange;
  text-shadow: 1px 1px 2px white,
    3px 3px 2px orangered;
}

.div--terminaluser {
  width: 90%;
  height: 75vh;
  margin: auto;
  display: flex;
  flex-direction: inline-block;

  background: transparent;
  border: 1px solid slategray;
  box-shadow: 0px 0px 4px slategray;
  border-radius: 15px 15px 15px 15px;

  .section--terminal {
    width: 80%;
    height: 100%;
    display: flex;
  }

  .div--worldbg {
    position: fixed;
    margin: auto;
    width: 90%;
    height: 75%;
    border-radius: 15px 15px 15px 15px;
    z-index: -1;
  }

  .img--bgterminal {
    border-radius: 15px 15px 15px 15px;
  }
}