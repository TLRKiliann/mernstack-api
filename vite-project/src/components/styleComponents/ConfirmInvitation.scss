$soft_blue: #93CAED;
$yellow: #ffff00;
$purple: #800080;
$sandybrown: #f4a460;
$orangered: #ff4500;
$whitesmoke: #f5f5f5;
$lightgrey_24: #242424;
$dark_moderate_red: #ae3c52;
$font_dejavu: 'DejaVu Sans', sans-serif;
$font_acme: 'Acme', sans-serif;

.box--confirminvitation {
  position: absolute;
  top: 0px;
  right: 0px;
  left: 0px;
  width: 500px;
  margin: auto;
  margin-top: 23%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  background: linear-gradient(180deg,
    $sandybrown, $purple);
  border-radius: 15px;
  z-index: 2;
  .confirm {
    margin: auto;
    margin-top: 20px;
    margin-bottom: 20px;
    font-family: $font_dejavu;
    font-size: 1.2rem;
    color: $purple;
    .span--confirmation {
      margin-right: 5px;
      margin-left: 5px;
      font-family: $font_acme;
      font-size: 1.8rem;
      text-shadow: 2px 1px 1px $purple;
      color: $soft_blue;
    }
  }
  .checkbox--div {
    width: 50%;
    margin: auto;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
  }
  .input--validationInvite {
    background-color: $lightgrey_24;
    color: $whitesmoke;
    border: 1px solid $yellow;
  }
  .btn--validationlast {
    margin-top: 20px;
    margin-bottom: 20px;
    padding: 5px 10px;
    font-family: $font_dejavu;
    font-size: 0.9rem;
    font-weight: bold;
    background: linear-gradient(180deg,
      $sandybrown, $orangered, $dark_moderate_red);
    border: 1px solid $purple;
    border-radius: 7px;
    color: $whitesmoke;
    &:hover {
      background: $sandybrown;
      color: $dark_moderate_red;
    }
  }
}