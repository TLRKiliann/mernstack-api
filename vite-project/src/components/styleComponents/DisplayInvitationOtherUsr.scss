.invitation--display {
  position: absolute;
  top: 0px;
  right: 0px;
  left: 0px;
  width: 400px;
  height: 400px;
  margin: auto;
  margin-top: 20%;

  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;

  background: linear-gradient(180deg,
    orange, purple);
  border-radius: 15px;
  z-index: 2;
}

.invitation--h4 {
  color: whitesmoke;
}

.btn--displayotheruser {
  margin-top: 20px;
  margin-bottom: 20px;
  padding: 5px 10px;
  font-family: "DejaVu Sans", sans-serif;
  font-size: 0.9rem;
  font-weight: bold;
  background: linear-gradient(180deg,
    orange, orangered, #ae3c52);
  border: 1px solid purple;
  border-radius: 7px;
  color: whitesmoke;
  &:hover {
    background: orange;
    color: #ae3c52;
  }
}