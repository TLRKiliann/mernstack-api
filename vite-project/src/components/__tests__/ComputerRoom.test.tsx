import React, {useState} from "react"
import { MemoryRouter, useParams } from 'react-router-dom'
import { screen, render } from '@testing-library/react'
import { test, expect, vi } from 'vitest'
import { create } from 'react-test-renderer'
import '@testing-library/jest-dom'
//import { useAuthLogin } from '../../context/AuthProvider'
import ComputerRoom from "../ComputerRoom"
import TerminalComponent from "../ComputerRoom"
import AskMessageBox from "../ComputerRoom"
import HandleAskUserOrigin from "../ComputerRoom"
import AskMessageBoxOrigin from "../ComputerRoom"
//import UsersOnline from "../ComputerRoom"
import DisplayInviteOrigin from "../ComputerRoom"
import DisplayInvitationOtherUsr from "../ComputerRoom"
import ConfirmationOrigin from "../ComputerRoom"
import ConfirmInvitation from "../ComputerRoom"

test("mocking UsersOnline", () => {
  vi.mock('../terminalchat/UsersOnline.tsx', () => ({
    UsersOnline: vi.fn()
  }))
})

test("0 Test useParams", () => {
  const mockUseParams = vi.fn()
  vi.mock('react-router-dom', () => ({
    ...vi.importActual('react-router-dom') as any,
    useParams: () => ({mockUseParams: "/"})
  }))
})

test('1 ComputerRoom is present', () => {
  render(
      <ComputerRoom />
  )
  const mytestid = screen.getByTestId("computerroomtest")
  expect(mytestid).toBeInTheDocument()
})

test('2 MatchSnapShot test ComputerRoom', () => {
  const tree = create(
      <ComputerRoom />
    )
  expect(tree.toJSON()).toMatchSnapshot()
})

test('3 ComputerRoom MatchSnapShot test ConfirmationOrigin', () => {
  const tree = create(
    <ConfirmationOrigin />
  )
  expect(tree.toJSON()).toMatchSnapshot()
})

test('4 ComputerRoom MatchSnapShot test ConfirmationOrigin', () => {
  const tree = create(
    <ConfirmationOrigin>
      <ConfirmInvitation />
    </ConfirmationOrigin>
  )
  expect(tree.toJSON()).toMatchSnapshot()
})

test('5 ComputerRoom MatchSnapShot test TerminalComponent', (roomStyle) => {
  const tree = create(
      <TerminalComponent roomStyle={roomStyle} />
  )
  expect(tree.toJSON()).toMatchSnapshot()
})

test('6 ComputerRoom MatchSnapShot test HandleAskUserOrigin', () => {
  const tree = create(
    <HandleAskUserOrigin />
  )
  expect(tree.toJSON()).toMatchSnapshot()
})

test('8 ComputerRoom MatchSnapShot test AskMessageBoxOrigin', () => {
  const tree = create(
    <AskMessageBoxOrigin />
  )
  expect(tree.toJSON()).toMatchSnapshot()
})

test('9 ComputerRoom MatchSnapShot test DisplayInviteOrigin', () => {
  const tree = create(
    <DisplayInviteOrigin />
  )
  expect(tree.toJSON()).toMatchSnapshot()
})

test('10 ComputerRoom MatchSnapShot test DisplayInviteOrigin', () => {
  const tree = create(
    <DisplayInviteOrigin>
      <DisplayInvitationOtherUsr />
    </DisplayInviteOrigin>
  )
  expect(tree.toJSON()).toMatchSnapshot()
})

//Correct
test('11 ComputerRoom MatchSnapShot test AskMessageBox', () => {
  const treeConfirm = create(
    <AskMessageBoxOrigin>
      <AskMessageBox />
    </AskMessageBoxOrigin>
  )
  expect(treeConfirm.toJSON()).toMatchSnapshot()
})

test("12 test func handleClose", () => {
  const beverage = {switchAsk: true}
  const handleClose = vi.fn(beverage => beverage)
  handleClose(beverage)
  expect(handleClose).toHaveReturned(true)
})


/*
const mockUseParams = vi.fn()
vi.mock('react-router-dom', () => ({
  ...vi.importActual('react-router-dom') as any,
  useParams: () => ({mockUseParams: "/"})
}))

const mockUseState = vi.fn()
vi.mock('react', () => ({
  ...vi.importActual('react') as any,
  useState: vi.fn()
}))

//expect to test createContext()
test("test ComputerRoom with method and spyOn", () => {
  const instance = ComputerRoom()
  vi.spyOn(instance, 'method')
})

const mockedAuth = vi.fn()
vi.mock('../../context/AuthProvider', () => ({
  ...vi.importActual('../../context/AuthProvider') as any,
  useAuthLogin: () => ({mockedAuth})
}))

test('7 ComputerRoom MatchSnapShot test HandleAskUserOrigin', () => {
    const tree = create(
      <HandleAskUserOrigin>
        <UsersOnline />
      </HandleAskUserOrigin>
    )
  expect(tree.toJSON()).toMatchSnapshot()
})
*/
