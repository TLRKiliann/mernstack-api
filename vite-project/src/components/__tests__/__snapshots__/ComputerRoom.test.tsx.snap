// Vitest Snapshot v1

exports[`2 MatchSnapShot test ComputerRoom 1`] = `
<div
  className="nextcomp--room"
  data-testid="computerroomtest"
>
  <div
    className="div--animationroomStyle"
  >
    <div
      className="rotation--roomstyle"
    >
      <h1 />
    </div>
  </div>
  <div />
  <h1
    className="title--room"
  >
    Room 
    /
  </h1>
  <div
    className="div--terminaluser"
  >
    <div
      className="section--terminal"
    >
      <div
        className="div--worldbg"
      >
        <img
          alt="/src/assets/background/world_connected.png"
          className="img--bgterminal"
          height="100%"
          src="/src/assets/background/world_connected.png"
          width="100%"
        />
      </div>
      <section
        className="terminal"
        data-testid="terminaltest"
      >
        <span
          className="intro--terminal"
        >
          <h3
            className="intro--terminalh3"
          />
        </span>
        <div
          className="div--terminal"
        />
        <div
          className="subterminal"
        >
          <div
            className="div--psubterminal"
          >
            <p
              className="arrow"
            >
              ▶
            </p>
          </div>
          <input
            className="input--message"
            onChange={[Function]}
            placeholder="Write something here..."
            required={true}
            type="text"
            value=""
          />
          <button
            className="btn--user"
            onClick={[Function]}
            type="button"
          >
            Enter
          </button>
        </div>
      </section>
    </div>
    <section
      className="user--online"
    >
      <div
        className="div--userolinetitle"
      >
        <h3
          className="userolinetitle"
        >
          Online Users
        </h3>
      </div>
    </section>
  </div>
</div>
`;

exports[`3 ComputerRoom MatchSnapShot test ConfirmationOrigin 1`] = `
<div
  className="nextcomp--room"
  data-testid="computerroomtest"
>
  <div
    className="div--animationroomStyle"
  >
    <div
      className="rotation--roomstyle"
    >
      <h1 />
    </div>
  </div>
  <div />
  <h1
    className="title--room"
  >
    Room 
    /
  </h1>
  <div
    className="div--terminaluser"
  >
    <div
      className="section--terminal"
    >
      <div
        className="div--worldbg"
      >
        <img
          alt="/src/assets/background/world_connected.png"
          className="img--bgterminal"
          height="100%"
          src="/src/assets/background/world_connected.png"
          width="100%"
        />
      </div>
      <section
        className="terminal"
        data-testid="terminaltest"
      >
        <span
          className="intro--terminal"
        >
          <h3
            className="intro--terminalh3"
          />
        </span>
        <div
          className="div--terminal"
        />
        <div
          className="subterminal"
        >
          <div
            className="div--psubterminal"
          >
            <p
              className="arrow"
            >
              ▶
            </p>
          </div>
          <input
            className="input--message"
            onChange={[Function]}
            placeholder="Write something here..."
            required={true}
            type="text"
            value=""
          />
          <button
            className="btn--user"
            onClick={[Function]}
            type="button"
          >
            Enter
          </button>
        </div>
      </section>
    </div>
    <section
      className="user--online"
    >
      <div
        className="div--userolinetitle"
      >
        <h3
          className="userolinetitle"
        >
          Online Users
        </h3>
      </div>
    </section>
  </div>
</div>
`;

exports[`4 ComputerRoom MatchSnapShot test ConfirmationOrigin 1`] = `
<div
  className="nextcomp--room"
  data-testid="computerroomtest"
>
  <div
    className="div--animationroomStyle"
  >
    <div
      className="rotation--roomstyle"
    >
      <h1 />
    </div>
  </div>
  <div />
  <h1
    className="title--room"
  >
    Room 
    /
  </h1>
  <div
    className="div--terminaluser"
  >
    <div
      className="section--terminal"
    >
      <div
        className="div--worldbg"
      >
        <img
          alt="/src/assets/background/world_connected.png"
          className="img--bgterminal"
          height="100%"
          src="/src/assets/background/world_connected.png"
          width="100%"
        />
      </div>
      <section
        className="terminal"
        data-testid="terminaltest"
      >
        <span
          className="intro--terminal"
        >
          <h3
            className="intro--terminalh3"
          />
        </span>
        <div
          className="div--terminal"
        />
        <div
          className="subterminal"
        >
          <div
            className="div--psubterminal"
          >
            <p
              className="arrow"
            >
              ▶
            </p>
          </div>
          <input
            className="input--message"
            onChange={[Function]}
            placeholder="Write something here..."
            required={true}
            type="text"
            value=""
          />
          <button
            className="btn--user"
            onClick={[Function]}
            type="button"
          >
            Enter
          </button>
        </div>
      </section>
    </div>
    <section
      className="user--online"
    >
      <div
        className="div--userolinetitle"
      >
        <h3
          className="userolinetitle"
        >
          Online Users
        </h3>
      </div>
    </section>
  </div>
</div>
`;

exports[`5 ComputerRoom MatchSnapShot test TerminalComponent 1`] = `
<div
  className="nextcomp--room"
  data-testid="computerroomtest"
>
  <div
    className="div--animationroomStyle"
  >
    <div
      className="rotation--roomstyle"
    >
      <h1 />
    </div>
  </div>
  <div />
  <h1
    className="title--room"
  >
    Room 
    /
  </h1>
  <div
    className="div--terminaluser"
  >
    <div
      className="section--terminal"
    >
      <div
        className="div--worldbg"
      >
        <img
          alt="/src/assets/background/world_connected.png"
          className="img--bgterminal"
          height="100%"
          src="/src/assets/background/world_connected.png"
          width="100%"
        />
      </div>
      <section
        className="terminal"
        data-testid="terminaltest"
      >
        <span
          className="intro--terminal"
        >
          <h3
            className="intro--terminalh3"
          />
        </span>
        <div
          className="div--terminal"
        />
        <div
          className="subterminal"
        >
          <div
            className="div--psubterminal"
          >
            <p
              className="arrow"
            >
              ▶
            </p>
          </div>
          <input
            className="input--message"
            onChange={[Function]}
            placeholder="Write something here..."
            required={true}
            type="text"
            value=""
          />
          <button
            className="btn--user"
            onClick={[Function]}
            type="button"
          >
            Enter
          </button>
        </div>
      </section>
    </div>
    <section
      className="user--online"
    >
      <div
        className="div--userolinetitle"
      >
        <h3
          className="userolinetitle"
        >
          Online Users
        </h3>
      </div>
    </section>
  </div>
</div>
`;

exports[`6 ComputerRoom MatchSnapShot test HandleAskUserOrigin 1`] = `
<div
  className="nextcomp--room"
  data-testid="computerroomtest"
>
  <div
    className="div--animationroomStyle"
  >
    <div
      className="rotation--roomstyle"
    >
      <h1 />
    </div>
  </div>
  <div />
  <h1
    className="title--room"
  >
    Room 
    /
  </h1>
  <div
    className="div--terminaluser"
  >
    <div
      className="section--terminal"
    >
      <div
        className="div--worldbg"
      >
        <img
          alt="/src/assets/background/world_connected.png"
          className="img--bgterminal"
          height="100%"
          src="/src/assets/background/world_connected.png"
          width="100%"
        />
      </div>
      <section
        className="terminal"
        data-testid="terminaltest"
      >
        <span
          className="intro--terminal"
        >
          <h3
            className="intro--terminalh3"
          />
        </span>
        <div
          className="div--terminal"
        />
        <div
          className="subterminal"
        >
          <div
            className="div--psubterminal"
          >
            <p
              className="arrow"
            >
              ▶
            </p>
          </div>
          <input
            className="input--message"
            onChange={[Function]}
            placeholder="Write something here..."
            required={true}
            type="text"
            value=""
          />
          <button
            className="btn--user"
            onClick={[Function]}
            type="button"
          >
            Enter
          </button>
        </div>
      </section>
    </div>
    <section
      className="user--online"
    >
      <div
        className="div--userolinetitle"
      >
        <h3
          className="userolinetitle"
        >
          Online Users
        </h3>
      </div>
    </section>
  </div>
</div>
`;

exports[`8 ComputerRoom MatchSnapShot test AskMessageBoxOrigin 1`] = `
<div
  className="nextcomp--room"
  data-testid="computerroomtest"
>
  <div
    className="div--animationroomStyle"
  >
    <div
      className="rotation--roomstyle"
    >
      <h1 />
    </div>
  </div>
  <div />
  <h1
    className="title--room"
  >
    Room 
    /
  </h1>
  <div
    className="div--terminaluser"
  >
    <div
      className="section--terminal"
    >
      <div
        className="div--worldbg"
      >
        <img
          alt="/src/assets/background/world_connected.png"
          className="img--bgterminal"
          height="100%"
          src="/src/assets/background/world_connected.png"
          width="100%"
        />
      </div>
      <section
        className="terminal"
        data-testid="terminaltest"
      >
        <span
          className="intro--terminal"
        >
          <h3
            className="intro--terminalh3"
          />
        </span>
        <div
          className="div--terminal"
        />
        <div
          className="subterminal"
        >
          <div
            className="div--psubterminal"
          >
            <p
              className="arrow"
            >
              ▶
            </p>
          </div>
          <input
            className="input--message"
            onChange={[Function]}
            placeholder="Write something here..."
            required={true}
            type="text"
            value=""
          />
          <button
            className="btn--user"
            onClick={[Function]}
            type="button"
          >
            Enter
          </button>
        </div>
      </section>
    </div>
    <section
      className="user--online"
    >
      <div
        className="div--userolinetitle"
      >
        <h3
          className="userolinetitle"
        >
          Online Users
        </h3>
      </div>
    </section>
  </div>
</div>
`;

exports[`9 ComputerRoom MatchSnapShot test DisplayInviteOrigin 1`] = `
<div
  className="nextcomp--room"
  data-testid="computerroomtest"
>
  <div
    className="div--animationroomStyle"
  >
    <div
      className="rotation--roomstyle"
    >
      <h1 />
    </div>
  </div>
  <div />
  <h1
    className="title--room"
  >
    Room 
    /
  </h1>
  <div
    className="div--terminaluser"
  >
    <div
      className="section--terminal"
    >
      <div
        className="div--worldbg"
      >
        <img
          alt="/src/assets/background/world_connected.png"
          className="img--bgterminal"
          height="100%"
          src="/src/assets/background/world_connected.png"
          width="100%"
        />
      </div>
      <section
        className="terminal"
        data-testid="terminaltest"
      >
        <span
          className="intro--terminal"
        >
          <h3
            className="intro--terminalh3"
          />
        </span>
        <div
          className="div--terminal"
        />
        <div
          className="subterminal"
        >
          <div
            className="div--psubterminal"
          >
            <p
              className="arrow"
            >
              ▶
            </p>
          </div>
          <input
            className="input--message"
            onChange={[Function]}
            placeholder="Write something here..."
            required={true}
            type="text"
            value=""
          />
          <button
            className="btn--user"
            onClick={[Function]}
            type="button"
          >
            Enter
          </button>
        </div>
      </section>
    </div>
    <section
      className="user--online"
    >
      <div
        className="div--userolinetitle"
      >
        <h3
          className="userolinetitle"
        >
          Online Users
        </h3>
      </div>
    </section>
  </div>
</div>
`;

exports[`10 ComputerRoom MatchSnapShot test DisplayInviteOrigin 1`] = `
<div
  className="nextcomp--room"
  data-testid="computerroomtest"
>
  <div
    className="div--animationroomStyle"
  >
    <div
      className="rotation--roomstyle"
    >
      <h1 />
    </div>
  </div>
  <div />
  <h1
    className="title--room"
  >
    Room 
    /
  </h1>
  <div
    className="div--terminaluser"
  >
    <div
      className="section--terminal"
    >
      <div
        className="div--worldbg"
      >
        <img
          alt="/src/assets/background/world_connected.png"
          className="img--bgterminal"
          height="100%"
          src="/src/assets/background/world_connected.png"
          width="100%"
        />
      </div>
      <section
        className="terminal"
        data-testid="terminaltest"
      >
        <span
          className="intro--terminal"
        >
          <h3
            className="intro--terminalh3"
          />
        </span>
        <div
          className="div--terminal"
        />
        <div
          className="subterminal"
        >
          <div
            className="div--psubterminal"
          >
            <p
              className="arrow"
            >
              ▶
            </p>
          </div>
          <input
            className="input--message"
            onChange={[Function]}
            placeholder="Write something here..."
            required={true}
            type="text"
            value=""
          />
          <button
            className="btn--user"
            onClick={[Function]}
            type="button"
          >
            Enter
          </button>
        </div>
      </section>
    </div>
    <section
      className="user--online"
    >
      <div
        className="div--userolinetitle"
      >
        <h3
          className="userolinetitle"
        >
          Online Users
        </h3>
      </div>
    </section>
  </div>
</div>
`;

exports[`11 ComputerRoom MatchSnapShot test AskMessageBox 1`] = `
<div
  className="nextcomp--room"
  data-testid="computerroomtest"
>
  <div
    className="div--animationroomStyle"
  >
    <div
      className="rotation--roomstyle"
    >
      <h1 />
    </div>
  </div>
  <div />
  <h1
    className="title--room"
  >
    Room 
    /
  </h1>
  <div
    className="div--terminaluser"
  >
    <div
      className="section--terminal"
    >
      <div
        className="div--worldbg"
      >
        <img
          alt="/src/assets/background/world_connected.png"
          className="img--bgterminal"
          height="100%"
          src="/src/assets/background/world_connected.png"
          width="100%"
        />
      </div>
      <section
        className="terminal"
        data-testid="terminaltest"
      >
        <span
          className="intro--terminal"
        >
          <h3
            className="intro--terminalh3"
          />
        </span>
        <div
          className="div--terminal"
        />
        <div
          className="subterminal"
        >
          <div
            className="div--psubterminal"
          >
            <p
              className="arrow"
            >
              ▶
            </p>
          </div>
          <input
            className="input--message"
            onChange={[Function]}
            placeholder="Write something here..."
            required={true}
            type="text"
            value=""
          />
          <button
            className="btn--user"
            onClick={[Function]}
            type="button"
          >
            Enter
          </button>
        </div>
      </section>
    </div>
    <section
      className="user--online"
    >
      <div
        className="div--userolinetitle"
      >
        <h3
          className="userolinetitle"
        >
          Online Users
        </h3>
      </div>
    </section>
  </div>
</div>
`;
